torch_glow/src/PyTorchModelLoader.cpp

```diff
@@ -33,69 +36,100 @@ inline glow::Type ptTypeToGlowType(const at::CompleteTensorType &ptType) {
+  return llvm::Expected<int32_t>(static_cast<int32_t>(val));
 }
 
 /// Given a Glow Node \p glowNode which must be a Glow Constant, returns a Glow
 /// Handle to that Constant's payload.
 template <typename T>
-const glow::Handle<T> getGlowConstantPayload(const glow::Node *glowNode) {
-  assert(glowNode);
-  const glow::Constant *glowConstant = llvm::dyn_cast<glow::Constant>(glowNode);
-  assert(glowConstant);
+llvm::Expected<glow::Handle<T>>
+getGlowConstantPayload(const glow::Node *glowNode) {
+  RETURN_ERR_IF_NOT(glowNode,
+                    glow::strFormat("Expected not null input glow node."));
+
+  const glow::Constant *glowConstant;
+  RETURN_ERR_IF_NOT(glowConstant = llvm::dyn_cast<glow::Constant>(glowNode),
+                    glow::strFormat("Expected glow node has Constant type."));
+
+  auto type = glowConstant->getElementType();
   if (std::is_same<int32_t, T>::value) {
-    assert(glowConstant->getElementType() == glow::ElemKind::Int32ITy);
+    RETURN_ERR_IF_NOT(type == glow::ElemKind::Int32ITy,
+                      glow::strFormat("Expected Int32ITy type, got %s",
+                                      glow::Type::getElementName(type).data()));
   } else if (std::is_same<float, T>::value) {
-    assert(glowConstant->getElementType() == glow::ElemKind::FloatTy);
+    RETURN_ERR_IF_NOT(type == glow::ElemKind::FloatTy,
+                      glow::strFormat("Expected FloatTy type, got %s",
+                                      glow::Type::getElementName(type).data()));
   } else if (std::is_same<bool, T>::value) {
-    assert(glowConstant->getElementType() == glow::ElemKind::BoolTy);
+    RETURN_ERR_IF_NOT(type == glow::ElemKind::BoolTy,
+                      glow::strFormat("Expected BoolTy type, got %s",
+                                      glow::Type::getElementName(type).data()));
   } else {
-    assert(false && "Unsupported type");
+    RETURN_ERR(glow::strFormat("Unsupported type %s",
+                               glow::Type::getElementName(type).data()));
   }
-  return glowConstant->getPayload().getHandle<T>();
+  return llvm::Expected<glow::Handle<T>>(
+      glowConstant->getPayload().getHandle<T>());
 }
 
 /// Given a Glow Handle \p handle and \p targetSize, will return an std::vector
 /// of the elements in the Tensor the Handle manages if the size of that Tensor
 /// is targetSize or if the Tensor has only one element in it, will replicate
 /// that element targetSize times.
-template <typename T, typename OutT = T>
-std::vector<OutT> expandParamIfNeeded(const glow::Handle<T> &handle,
-                                      size_t targetSize) {
-  assert(handle.dims().size() == 1 && "Expected a 1d Glow Tensor");
-  assert(handle.size() == 1 ||
-         handle.size() == targetSize &&
-             "Expected input size to be either 1 or the target size");
-  std::vector<OutT> out;
+template <typename T, typename OutT>
+llvm::Error expandParamIfNeeded(const glow::Handle<T> &handle,
```

Comment:
Any reason to not just use Expected<OutT>?

Reply:
Template method usually deduces template types from input/output parameters, therefore sending output template parameter explicitly (not as a return value) makes compiler happy.